local-cluster/tests/local_cluster.rs

```diff
@@ -1431,12 +1438,6 @@ fn test_optimistic_confirmation_violation_with_no_tower() {
             prev_voted_slot
         );
         blockstore.set_dead_slot(prev_voted_slot).unwrap();
-
-        std::fs::remove_file(Tower::get_filename(
```

Comment:
Is this no longer necessary?

Reply:
@carllin Yeah. As I figured out this in hard way, saved tower doesn't protect us from violating opt. conf., _if voted slot is marked dead, forcibly_. So, as you guessed, i'm effectively making this test harder to pass (=actually violate) by not removing the saved tower and indeed this test still passes. :)

https://github.com/solana-labs/solana/pull/10718#discussion_r490104064:
> Well, I found it's very hard to make a validator violate optimistic confirmation _if and only if tower is removed_.
> The hard part is that marking a slot as dead makes a validator to violate the optimistic conf. **even if tower is persited**. And eventually, the cluster makes new roots.
> ...

Reply:
I added explicit comment mentioning for the saved tower: https://github.com/solana-labs/solana/pull/12350/files#r506909261